In [1]:
import numpy as np
import pandas as pd

In [2]:
def check(filename):
    n_lots = 0
    lots = {}
    data = pd.read_csv(filename)
    for i in data.index:
        lot, p = data['ID Лота'][i], data['Цена'][i]
        if lot in lots:
            lots[lot] += [p, 1]
        else:
            lots[lot] = [p, 1]
            n_lots += 1
    p_mean = 0
    for l, items in lots.items():
        p_mean += items[0]/items[1]
    p_mean /= n_lots
    return n_lots, p_mean

In [8]:
def quality(their, ours):
    t_lots, t_mean = check(their) 
    o_lots, o_mean = check(ours)
    print(((1-o_lots/t_lots)+(1-o_mean/t_mean))/2)

In [23]:
quality('./Data/2020-3.csv', './Result/2020-3.csv')

0.7847191023101884


Качество: 0.5927475285473656
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-4

Качество: 0.6218961400185278
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-2

Качество: 0.6218961400185278
Параметры: prod_percent = 50, prov_percent = 50, podgon = 0.5

Качество: 0.7316381224550773
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-6     !!!!!!!

MQ: 0.7847191023101884
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-8     !!!!!!!

Качество: 0.7347552206818162
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-10

Качество: 0.614426300673727
Параметры: prod_percent = 60, prov_percent = 60, podgon = 1e-6

Качество: 0.6171318247265578
Параметры: prod_percent = 70, prov_percent = 70, podgon = 1e-6

Качество: 0.6171318247265578
Параметры: prod_percent = 80, prov_percent = 80, podgon = 1e-6

Качество: 0.7316381224550773
Параметры: prod_percent = 40, prov_percent = 40, podgon = 1e-6

Качество: 0.7316381224550773
Параметры: prod_percent = 30, prov_percent = 30, podgon = 1e-6

Качество: 0.6438936676388944 
Параметры: prod_percent = 40, prov_percent = 40, podgon = 1e-7

Качество: 0.7796388632205948
Параметры: prod_percent = 40, prov_percent = 40, podgon =  1e-9

Качество: 0.7796388632205948
Параметры: prod_percent = 50, prov_percent = 50, podgon = 1e-9

Качество: 0.6623030594139343
Параметры: prod_percent = 50, prov_percent = 50, podgon = 15 * 1e-9

Качество: 
Параметры: prod_percent = 50, prov_percent = 50, podgon = 

Качество: 
Параметры: prod_percent = 50, prov_percent = 50, podgon = 

Качество: 
Параметры: prod_percent = 50, prov_percent = 50, podgon = 